packages/core/src/layout/Sidebar/Items.tsx

```diff
@@ -19,91 +19,204 @@ import {
+  TextField,
   Theme,
   Typography,
+  Badge,
 } from '@material-ui/core';
+import SearchIcon from '@material-ui/icons/Search';
 import clsx from 'clsx';
-import React, { FC, useContext } from 'react';
+import React, { FC, useContext, useState, KeyboardEventHandler } from 'react';
 import { sidebarConfig, SidebarContext } from './config';
 
-const useStyles = makeStyles<Theme>(theme => ({
-  root: {
-    color: '#b5b5b5',
-    display: 'flex',
-    flexFlow: 'row nowrap',
-    alignItems: 'center',
-    height: 40,
-    cursor: 'pointer',
-  },
-  closed: {
-    width: sidebarConfig.drawerWidthClosed,
-    justifyContent: 'center',
-  },
-  open: {
-    width: sidebarConfig.drawerWidthOpen,
-  },
-  label: {
-    fontWeight: 'bolder',
-    whiteSpace: 'nowrap',
-    lineHeight: 1.0,
-    marginLeft: theme.spacing(1),
-  },
-  iconContainer: {
-    height: '100%',
-    width: sidebarConfig.drawerWidthClosed,
-    marginRight: -theme.spacing(2),
-    display: 'flex',
-    alignItems: 'center',
-    justifyContent: 'center',
-  },
-}));
+const useStyles = makeStyles<Theme>(theme => {
+  const {
+    selectedIndicatorWidth,
+    drawerWidthClosed,
+    drawerWidthOpen,
+    iconContainerWidth,
+    iconSize,
+  } = sidebarConfig;
+
+  return {
+    root: {
+      color: '#b5b5b5',
+      display: 'flex',
+      flexFlow: 'row nowrap',
+      alignItems: 'center',
+      height: 48,
+      cursor: 'pointer',
+      '&$selected': {
+        borderLeft: `solid ${selectedIndicatorWidth}px #9BF0E1`,
+      },
+    },
+    closed: {
+      width: drawerWidthClosed,
+      justifyContent: 'center',
+      '&$selected': {
+        width: drawerWidthClosed - selectedIndicatorWidth,
+      },
+    },
+    open: {
+      width: drawerWidthOpen,
+    },
+    label: {
+      // XXX (@koroeskohr): I can't seem to achieve the desired font-weight from the designs
+      fontWeight: 'bold',
+      whiteSpace: 'nowrap',
+      lineHeight: 1.0,
+    },
+    iconContainer: {
+      boxSizing: 'border-box',
+      height: '100%',
+      width: iconContainerWidth,
+      marginRight: -theme.spacing(2),
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      '&$selected': {
+        marginLeft: -selectedIndicatorWidth,
+      },
+    },
+    icon: {
+      width: iconSize,
+      height: iconSize,
+      color: '#b5b5b5',
+      '&$selected': {
+        color: '#ffffff',
+      },
+    },
+    searchRoot: {
+      marginBottom: 12,
+    },
+    searchField: {
+      color: '#b5b5b5',
+      fontWeight: 'bold',
+      fontSize: theme.typography.fontSize,
+    },
+    searchContainer: {
+      width: drawerWidthOpen - iconContainerWidth,
+    },
+    selected: {},
+  };
+});
 
 type SidebarItemProps = {
   icon: typeof SvgIcon;
-  text: string;
+  text?: string;
   to?: string;
+  selected?: boolean;
   onClick?: () => void;
 };
 
 export const SidebarItem: FC<SidebarItemProps> = ({
   icon: Icon,
   text,
   to,
+  selected,
```

Comment:
Not sure when this would be used tbh

Reply:
This is used for styling basically, duplicating style was way more bothersome